src/flash/flasher.h

```diff
@@ -0,0 +1,95 @@
+#pragma once
+
+#include "linkconfiguration.h"
+
+#include <QLoggingCategory>
+#include <QProcess>
+
+
+Q_DECLARE_LOGGING_CATEGORY(FLASH)
+
+/**
+ * @brief Manage the project Flasher
+ *
+ */
+class Flasher : public QObject
+{
+    Q_OBJECT
+public:
+    /**
+     * @brief Construct a new Network Manager object
+     *
+     */
+    Flasher(QObject* parent = nullptr);
+
+    /**
+     * @brief Destroy the Flash Manager object
+     *
+     */
+    ~Flasher() = default;
+
+    /**
+     * @brief Start the flash procedure
+     *
+     */
+    void flash();
+
+    /**
+     * @brief Set the flash baud rate
+     *
+     * @param baudRate
+     */
+    bool setBaudRate(int baudRate);
+
+    /**
+     * @brief Set the binary path
+     *
+     * @param firmwareFilePath
+     */
+    bool setFirmwarePath(const QString& firmwareFilePath);
+
+    /**
+     * @brief Set the flash link
+     *
+     * @param link
+     */
+    bool setLink(const LinkConfiguration& link);
+
+    /**
+     * @brief Enable/disable verify
+     *
+     * @param verify
+     */
+    void setVerify(bool verify);
+
+    /**
+     * @brief Defines flash state
+     *
+     */
+    enum States {
```

Comment:
It looks like you've mixed flasher state and flash process result, suggest two enums: FlasherState, and FlashProcessResult.

Reply:
Removed unused errors enums and created a single enum and an error propriety.